### PR TITLE
Servlets

### PR DESCRIPTION
**Pull request by feeL67**
Add Servlets Homework

### CODE REVIEW

File: src/main/java/com/tms/web/CarsServlet.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CarsServlet extends HttpServlet {
+
+    Map<String, String> cars = new HashMap<>();
```
Comment (gr00shik) [inline src/main/java/com/tms/web/CarsServlet.java:14]: no, it is a rule - don't use state inside servlets

File: src/main/java/com/tms/web/CarsServlet.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String id = req.getParameter("id");
+        String mark = req.getParameter("mark");
+        PrintWriter writer = resp.getWriter();
+        if (cars.containsKey(id)) {
+            writer.println("<html><body><h2> Car with this id " + id + "- exist!<h2></body></html>");
+        } else if (id != null & mark != null) {
```
Comment (gr00shik) [inline src/main/java/com/tms/web/CarsServlet.java:34]: this logic should be in service layer

File: src/main/java/com/tms/web/MyListener.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletRequestEvent;
+import javax.servlet.ServletRequestListener;
+import javax.servlet.annotation.WebListener;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+@WebListener("/*")
```
Comment (gr00shik) [inline src/main/java/com/tms/web/MyListener.java:11]: listener reacts on event, don't use pattern

**Review by gr00shik — COMMENTED** [review]
ok

File: src/main/java/com/tms/web/MyListener.java (hunk 3 above)
Comment (feeL67) [reply src/main/java/com/tms/web/MyListener.java:11]: ok

File: src/main/java/com/tms/web/CarsServlet.java (hunk 2 above)
Comment (feeL67) [reply src/main/java/com/tms/web/CarsServlet.java:34]: accepted

File: src/main/java/com/tms/web/CarsServlet.java (hunk 1 above)
Comment (feeL67) [reply src/main/java/com/tms/web/CarsServlet.java:14]: ok

**Pull request merged**